src/Workspaces/Core/Portable/FlowAnalysis/LValueFlowCaptureProvider.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+    /// <summary>
+    /// Helper class to detect <see cref="IFlowCaptureOperation"/>s that are l-value captures.
+    /// L-value captures are essentially captures of a symbol's location/address.
+    /// Corresponding <see cref="IFlowCaptureReferenceOperation"/>s which share the same
+    /// <see cref="CaptureId"/> as this flow capture, dereferences and writes to this locaion
+    /// subsequently in the flow graph.
+    /// For example, consider the below code:
+    ///     a[i] = x ?? a[j];
+    /// The control flow graph contains an initial flow capture of "a[i]" to capture the l-value
+    /// of this array element:
+    ///     FC0 (a[i])
+    /// Then it evaluates the right hand side, which can have different
+    /// values on different control flow paths, and the resultant value is then written
+    /// to the captured location:
+    ///     FCR0 = result
+    /// </summary>
+    /// <remarks>
+    /// NOTE: This type is a temporary workaround for https://github.com/dotnet/roslyn/issues/31007
+    /// and it can be deleted once that feature is implemented.
+    /// </remarks>
```

Comment:
so, for example, if i had:    `... ? ref a[i] : ref a[j]` there would be two lvalue captures there?

Reply:
I think this unit test should clarify your question: [http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.IOperation.UnitTests/IOperation/IOperationTests_ITupleExpression.cs,1693](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.IOperation.UnitTests/IOperation/IOperationTests_ITupleExpression.cs,1693). 

---
In reply to: [231667469](https://github.com/dotnet/roslyn/pull/31025#discussion_r231667469) [](ancestors = 231667469)